fetcher/fetcher/parse.py

```diff
@@ -22,60 +23,97 @@
+
+
+class ParsedFile(BaseModel):
+    feed_type: Union[FeedType, GtfsScheduleFileType]
+    hash: bytes
+    records: Iterable[Dict]
 
 
-def file_to_records(file: RawFetchedFile) -> Iterable[Tuple[Union[FeedType, GtfsScheduleFileType], Iterable[Dict]]]:
+def file_to_records(
+    file: RawFetchedFile,
+) -> Iterable[ParsedFile]:
     pydantic_type = FEED_TYPES[file.config.feed_type]
     try:
         if file.config.feed_type == FeedType.gtfs_schedule:
             with zipfile.ZipFile(BytesIO(file.contents)) as zipf:
                 for zipf_file in zipf.namelist():
-                    # https://stackoverflow.com/a/60918920
-                    with io.TextIOWrapper(zipf.open(zipf_file), encoding="utf-8") as f:
-                        reader = csv.DictReader(f)
-                        # TODO: this will throw an error if attempting to parse a file we don't enumerate
-                        #  we probably want to just throw a warning/generate an outcome rather than stop
-                        #  further processing
-                        # looking up the enum by value not name
-                        yield GtfsScheduleFileType(zipf_file), parse_obj_as(ListOfDicts, list(reader)).records
+                    with zipf.open(zipf_file) as f:
+                        contents = f.read()
+                    reader = csv.DictReader(io.TextIOWrapper(io.BytesIO(contents), encoding="utf-8"))
+                    # TODO: this will throw an error if attempting to parse a file we don't enumerate
+                    #  we probably want to just throw a warning/generate an outcome rather than stop
+                    #  further processing
+                    # looking up the enum by value not name
+                    yield ParsedFile(
+                        feed_type=GtfsScheduleFileType(zipf_file),
+                        hash=hashlib.md5(contents).digest(),
+                        records=parse_obj_as(ListOfDicts, list(reader)).records,
+                    )
         elif pydantic_type == GtfsRealtime:
             feed = gtfs_realtime_pb2.FeedMessage()
             feed.ParseFromString(file.contents)
-            yield file.config.feed_type, GtfsRealtime(**MessageToDict(feed)).records
+            yield ParsedFile(
+                feed_type=file.config.feed_type,
+                hash=hashlib.md5(file.contents).digest(),
+                records=GtfsRealtime(**MessageToDict(feed)).records,
+            )
         else:
-            yield file.config.feed_type, parse_obj_as(pydantic_type, json.loads(file.contents)).records
+            yield ParsedFile(
+                feed_type=file.config.feed_type,
+                hash=hashlib.md5(file.contents).digest(),
+                records=parse_obj_as(pydantic_type, json.loads(file.contents)).records,
+            )
     except (ValidationError, DecodeError) as e:
         typer.secho(f"{type(e)} occurred on {file.bucket}/{file.gcs_key}", fg=typer.colors.RED)
         raise
```

Comment:
Wait sorry so if we get an invalid RT proto like we did from SEPTA a few days ago, does that halt the whole parse job? What else will not get parsed as a result? 🤔 Like does it go in a deterministic order?

Reply:
yes but we haven't hit this; I want to explicitly add error handling as we encounter the conditions